go/cmd/server/logging/logging_test.go

```diff
@@ -0,0 +1,223 @@
+package logging
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/feast-dev/feast/go/internal/feast/model"
+	"github.com/feast-dev/feast/go/internal/test"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestLoggingChannelTimeout(t *testing.T) {
+	// Pregenerated using `feast init`.
+	loggingService, err := NewLoggingService(nil, 1, "", false)
+	assert.Nil(t, err)
+	assert.Empty(t, loggingService.memoryBuffer.logs)
+	ts := timestamppb.New(time.Now())
+	newLog := Log{
+		FeatureStatuses: []serving.FieldStatus{serving.FieldStatus_PRESENT},
+		EventTimestamps: []*timestamppb.Timestamp{ts, ts},
+	}
+	loggingService.EmitLog(&newLog)
+	// Wait for memory buffer flush
+	time.Sleep(20 * time.Millisecond)
+	newTs := timestamppb.New(time.Now())
+
+	newLog2 := Log{
+		FeatureStatuses: []serving.FieldStatus{serving.FieldStatus_PRESENT},
+		EventTimestamps: []*timestamppb.Timestamp{newTs, newTs},
+	}
+	err = loggingService.EmitLog(&newLog2)
+	// The channel times out and doesn't hang.
+	time.Sleep(20 * time.Millisecond)
+	assert.NotNil(t, err)
+}
+
+func TestSchemaTypeRetrieval(t *testing.T) {
```

Comment:
Taking into account comments above this test is not enough. We need better coverage of schema generation